### ISSUE TITLE
Automatic resubmit does not work on Edison

### ISSUE THREAD

**Issue by xuezhengllnl** [E3SM-Project/E3SM#2868]
The hash number is: 4a3194ddd0cb16cedfd03eb23f2c4d82ea66d797
Here is the error message:
```
 45 2019-04-15 18:33:53 SAVE_POSTRUN_PROVENANCE HAS FINISHED
 46 check for resubmit
 47 dout_s False
 48 mach edison
 49 resubmit_num 5
 50 Traceback (most recent call last):
 51   File "/var/spool/slurmd/job12981979/slurm_script", line 84, in <module>
 52     _main_func(__doc__)
 53   File "/var/spool/slurmd/job12981979/slurm_script", line 79, in _main_func
 54     success = case.case_run(skip_pnl=skip_pnl, set_continue_run=set_continue_run, submit_resubmits=resubmit)
 55   File "/global/cscratch1/sd/xzheng/E3SM_code/20190325-master/cime/scripts/Tools/../../scripts/lib/CIME/case/case_run.py", line 333, in case_run
 56     _resubmit_check(self)
 57   File "/global/cscratch1/sd/xzheng/E3SM_code/20190325-master/cime/scripts/Tools/../../scripts/lib/CIME/case/case_run.py", line 242, in _resubmit_check
 58     case.submit(job=job, resubmit=True)
 59   File "/global/cscratch1/sd/xzheng/E3SM_code/20190325-master/cime/scripts/Tools/../../scripts/lib/CIME/case/case_submit.py", line 182, in submit
 60     batch_args = config.get('SubmitOptions', 'batch_args')
 61   File "/global/common/edison/software/python/2.7-anaconda-4.4/lib/python2.7/ConfigParser.py", line 623, in get
 62     return self._interpolate(section, option, value, d)
 63   File "/global/common/edison/software/python/2.7-anaconda-4.4/lib/python2.7/ConfigParser.py", line 691, in _interpolate
 64     self._interpolate_some(option, L, rawval, section, vars, 1)
 65   File "/global/common/edison/software/python/2.7-anaconda-4.4/lib/python2.7 /ConfigParser.py", line 732, in _interpolate_some
 66     "'%%' must be followed by '%%' or '(', found: %r" % (rest,))
 67 ConfigParser.InterpolationSyntaxError: '%' must be followed by '%' or '(', found: '%j'
```

**Comment by rljacob** [comment]
Can you see if latest master has this problem?

**Comment by xuezhengllnl** [comment]
> Can you see if latest master has this problem?
@rljacob I tried with the latest master and it had the same error message (batch_output/edison.20190419-master.Test_resub.ne30_ne30.o13017481):
 48 mach edison
 49 resubmit_num 5
 50 Traceback (most recent call last):
 51   File "/var/spool/slurmd/job13017481/slurm_script", line 84, in     <module>
 52     _main_func(__doc__)
 53   File "/var/spool/slurmd/job13017481/slurm_script", line 79, in     _main_func
 54     success = case.case_run(skip_pnl=skip_pnl, set_continue_run=s    et_continue_run, submit_resubmits=resubmit)
 55   File "/global/cscratch1/sd/xzheng/E3SM_code/20190419-master/cim    e/scripts/Tools/../../scripts/lib/CIME/case/case_run.py", line 33    3, in case_run
 56     _resubmit_check(self)
 57   File "/global/cscratch1/sd/xzheng/E3SM_code/20190419-master/cim    e/scripts/Tools/../../scripts/lib/CIME/case/case_run.py", line 24    2, in _resubmit_check
 58     case.submit(job=job, resubmit=True)
 59   File "/global/cscratch1/sd/xzheng/E3SM_code/20190419-master/cim    e/scripts/Tools/../../scripts/lib/CIME/case/case_submit.py", line     187, in submit
 60     batch_args = config.get('SubmitOptions', 'batch_args')
 61   File "/global/common/edison/software/python/2.7-anaconda-4.4/li    b/python2.7/ConfigParser.py", line 623, in get
 62     return self._interpolate(section, option, value, d)
 63   File "/global/common/edison/software/python/2.7-anaconda-4.4/li    b/python2.7/ConfigParser.py", line 691, in _interpolate
 64     self._interpolate_some(option, L, rawval, section, vars, 1)
 65   File "/global/common/edison/software/python/2.7-anaconda-4.4/li    b/python2.7/ConfigParser.py", line 732, in _interpolate_some
 66     "'%%' must be followed by '%%' or '(', found: %r" % (rest,))
 67 ConfigParser.InterpolationSyntaxError: '%' must be followed by '%    ' or '(', found: '%j'

**Comment by rljacob** [comment]
One more thing, what is the create_newcase command you are using to set this up?

**Comment by jgfouca** [comment]
Passing along to Joe.

**Comment by milenaveneziani** [comment]
A quick note to say that I am using hash a3452b7a7 and the latest e3sm-unified env on edison, and the RESUBMIT option works for me.

**Comment by xuezhengllnl** [comment]
> One more thing, what is the create_newcase command you are using to set this up?

2019-04-19 13:52:57: /global/cscratch1/sd/xzheng/E3SM_code/20190419-master/cime/scripts/create_newcase --case edison.20190419-master.Test_resub.ne30_ne30 --compset FC5AV1C-04P2 --script-root /global/cscratch1/sd/xzheng/E3SM_simulations/edison.20190419-master.Test_resub.ne30_ne30/case_scripts --res ne30_ne30 --pecount S --handle-preexisting-dirs u --mach edison --output-root /global/cscratch1/sd/xzheng/E3SM_simulations/edison.20190419-master.Test_resub.ne30_ne30 --project acme

**Comment by jhkennedy** [comment]
Update: This issue was introduced to CIME in ESMCI/cime#2518 , and into E3SM on the next CIME update. 

It boils down to: 

NERSC slurm batch system config (`cime/config/e3sm/machines/config_machines.xml`) includes a batch directive like: 
```xml
      <directive> --output={{ job_id }}.%j </directive>
```

Which for your case get's saved to a `.submit_options` file in the case directory like: 
```ini
[SubmitOptions]
batch_args =  --job-name=Test_resub --output=batch_output/edison.20190419-master.Test_resub.ne30_ne30.o%j 
```
Then, when resubmit happens, python's `configparser.SafeConfigParser` (just `configparser.ConfigParser` in py3) is used to load the `batch_args` variable from the `.submit_options` file. However, `(Safe)ConfigParser` uses `%` to do magic variable interpolation, and tries to interpolate the  `%j` in that string. 

*Note: This likely affects all E3SM machines using the batch systems `lsf`, `nersc_slurm`, or `slurm`, and all CESM machines using the batch system `lsf`, as they all have a `%` in their output directives*


The fix is to either use the `RawConfigParser` class (which doesn't do interpolation) or to escape the `%` with a `%%` as we write to disk -- if interpolation *is* being used, that second option may get tricky to determine which `%` need to be escaped vs not. 

Since this effects CESM as well, I'm going to open an issue on ESMCI/cime to resolve this.

If this is blocking anyone's workflow, you can apply this patch to fix it:

```diff
diff --git a/cime/scripts/lib/CIME/case/case_submit.py b/cime/scripts/lib/CIME/case/case_submit.py
index 9907f6b7ab..0f6e9cd4af 100644
--- a/cime/scripts/lib/CIME/case/case_submit.py
+++ b/cime/scripts/lib/CIME/case/case_submit.py
@@ -175,7 +175,7 @@ def submit(self, job=None, no_batch=False, prereq=None, allow_fail=False, resubm
     # any submit options used on the original submit and use them again
     submit_options = os.path.join(caseroot, ".submit_options")
     if resubmit and os.path.exists(submit_options):
-        config = configparser.SafeConfigParser()
+        config = configparser.RawConfigParser()
         config.read(submit_options)
         if not skip_pnl and config.has_option('SubmitOptions','skip_pnl'):
             skip_pnl = config.getboolean('SubmitOptions', 'skip_pnl')
```

**Comment by xylar** [comment]
`SafeConfigParser` is also depricated as of python 3.2 and going away in 3.8 so that's another (smaller) reason to make the change above.

**Comment by jhkennedy** [comment]
@xylar , in Python 3.2+ both `SafeConfigParser` and `RawConfigParser` are aliases to `ConfigParaser` with different `interpolation=` values, and kept for legacy code reasons. `SafeConfigParser` appears to have been undocumented since Python 3.2, but it's good to know it won't be available as of 3.8! Looks like `RawConfigParser` is going to stick around for a while though.

Really, once Python 2 isn't supported, this should be converted to `ConfigParser(interpolation=None)`

**Comment by jhkennedy** [comment]
Now on master.
